noir/noir-repo/test_programs/rebuild.sh

```diff
@@ -60,7 +62,38 @@ for dir in $current_dir/benchmarks/*; do
     dirs_to_process+=("$dir")
 done
 
+pids=() # Array to hold PIDs of background processes
+dirs_map=() # Array to map PIDs to directories
+
+if [ -z $NO_PARALLEL ]; then
+    # Process directories in parallel
+    for dir in "${dirs_to_process[@]}"; do
+        process_dir "$dir" "$current_dir" &  # Run process_dir in the background
+        pid=$!  # Get PID of the last background command
+        pids+=($pid)  # Add PID to the pids array
+        dirs_map[$pid]=$dir  # Map PID to the directory being processed
+    done
+else
+    # Process directories sequentially
+    for dir in "${dirs_to_process[@]}"; do
+        process_dir "$dir" "$current_dir"  # Run process_dir in the foreground
+        pid=$!  # Get PID of the last command
+        pids+=($pid)  # Add PID to the pids array
+        dirs_map[$pid]=$dir  # Map PID to the directory being processed
+    done
+fi
+
+# Check the exit status of each background job.
+for pid in "${pids[@]}"; do
+    if ! wait $pid; then  # Wait for the process to complete, check if it failed
+        echo "Rebuild failed for directory: ${dirs_map[$pid]}"  # Print failed directory
+        exit_status=$?  # Capture the failed exit status
```

Comment:
Won't this just get the last exit status? Should we just exit here? This is nit though and otherwise updates look good

Reply:
oh yeah, I think I did correct this but I must've copy pasted the wrong version

Reply:
also, ! wait $pid should only be true if the exit status failed so its fine.
Shouldn't just exit here since I wanted to print all the failing ones.

This actually gets the exit status of the echo statement, so I need to move this line above

Reply:
It looks like we now loop over all jobs and the failed pids. I feel we could still print failed jobs while looping over all the jobs but what you did keeps the script simple and we ideally shouldn't have that many failed jobs.